### PR TITLE
Fix signed integer overflow

### PR DESCRIPTION
**Pull request by shao-hua-li**
related to https://github.com/libsndfile/libsndfile/issues/785

### CODE REVIEW

**Comment by evpobr** [comment]
@shao-hua-li ?

File: src/gsm610.c [hunk 1]
```diff
@@ -150,7 +150,7 @@ Need separate gsm_data structs for encode and decode.
 			pgsm610->blocks = psf->datalength / pgsm610->blocksize + 1 ;
 			} ;
 
-		psf->sf.frames = pgsm610->samplesperblock * pgsm610->blocks ;
+		psf->sf.frames = (long)pgsm610->samplesperblock * pgsm610->blocks ;
```
Comment (evpobr) [inline src/gsm610.c:153]:
```suggestion
		psf->sf.frames = (sf_count_t) pgsm610->samplesperblock * pgsm610->blocks ;
```

**Review by evpobr — CHANGES_REQUESTED** [review]

File: src/gsm610.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (shao-hua-li) [reply src/gsm610.c:153]: Yes, I agree with you.

**Review by evpobr — APPROVED** [review]

**Comment by evpobr** [comment]
Okay, let's wait a couple of days so that the rest of the maintainers have time to see it.

**Review by arthurt — APPROVED** [review]
Regardless of #785, this change does have merit.

**Pull request merged**